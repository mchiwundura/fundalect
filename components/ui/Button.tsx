import { StyleSheet, TouchableOpacity } from "react-native";
import React from "react";
import { ThemedText } from "../ThemedText";
import { Colors } from "@/constants/Colors";
import { ButtonProps } from "@/types/ui/Button";

const Button = ({ onPress, title, style, type }: ButtonProps) => {
  return (
    <TouchableOpacity
      style={[
        styles.container,
        type === "primary" && styles.primaryContainer,
        style,
      ]}
      onPress={onPress}
    >
      <ThemedText type="subtitle">{title}</ThemedText>
    </TouchableOpacity>
  );
};

export default Button;

const styles = StyleSheet.create({
  container: {
    marginVertical: 12,
    maxWidth: 360,
    width: "100%",
    height: 40,
    borderRadius: 10,
    justifyContent: "center",
    alignItems: "center",
  },
  primaryContainer: {
    backgroundColor: Colors.primary,
  },
});
