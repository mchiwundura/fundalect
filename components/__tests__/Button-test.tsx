import { render } from "@testing-library/react-native";
import Button from "../ui/Button";

describe("Button Component Test", () => {
  it("renders correctly as default", () => {
    const tree = render(
      <Button onPress={() => {}} title="Testing Primary" />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with primary type", () => {
    const tree = render(
      <Button type="primary" onPress={() => {}} title="Testing Primary" />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
