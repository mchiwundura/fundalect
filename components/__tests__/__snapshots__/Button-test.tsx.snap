// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Test renders correctly as default 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderRadius": 10,
      "height": 40,
      "justifyContent": "center",
      "marginVertical": 12,
      "maxWidth": 360,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "rgba(0,0,0,0.8)",
        },
        undefined,
        undefined,
        undefined,
        {
          "fontSize": 20,
          "fontWeight": "bold",
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Testing Primary
  </Text>
</View>
`;

exports[`Button Component Test renders correctly with primary type 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#9584FF",
      "borderRadius": 10,
      "height": 40,
      "justifyContent": "center",
      "marginVertical": 12,
      "maxWidth": 360,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "rgba(0,0,0,0.8)",
        },
        undefined,
        undefined,
        undefined,
        {
          "fontSize": 20,
          "fontWeight": "bold",
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Testing Primary
  </Text>
</View>
`;
